istio-proxy/fetch.sh

```diff
@@ -2,28 +2,56 @@ set -x
 set -e
 
 PROXY_DIR=${PROXY_DIR:-istio-proxy}
+
 function check_envs() {
   if [ -z "$FETCH_DIR" ]; then
     echo "FETCH_DIR required. Please set"
     exit 1
   fi
 
-	PROXY_FETCH_DIR=${FETCH_DIR}/${PROXY_DIR}
+  PROXY_FETCH_DIR=${FETCH_DIR}/${PROXY_DIR}
   CACHE_DIR=${PROXY_FETCH_DIR}/bazel
 }
 
+function check_git_hashes() {
+  local commit_hash_name="$1"
+  local tarball_hash_name="$2"
+  eval commit_hash=\$$commit_hash_name
+  eval tarball_hash=\$$tarball_hash_name
+
+  if [ ! -z "${commit_hash}" ] || [ ! -z "${tarball_hash}" ]; then
+    if [ -z "${commit_hash}" ]; then
+      echo "ERROR: ${tarball_hash_name} was set but missing ${commit_hash_name}";
+      exit 1;
+    fi
+
+    if [ -z "${tarball_hash}" ]; then
+      echo "ERROR: ${commit_hash_name} was set but missing ${tarball_hash_name}";
+      exit 1;
+    fi
+  fi
+}
+
 function set_default_envs() {
   if [ -z "${PROXY_GIT_REPO}" ]; then
     PROXY_GIT_REPO=https://github.com/maistra/proxy
   fi
+  check_git_hashes "PROXY_GIT_COMMIT_HASH" "PROXY_TARBALL_SHA256"
+
+  if [ -z "${ISTIO_PROXY_OPENSSL_GIT_REPO}" ]; then
+    ISTIO_PROXY_OPENSSL_GIT_REPO=https://github.com/maistra/istio-proxy-openssl
+  fi
```

Comment:
no need to change that now, but for things like this we can make it simpler with:
```
: ${ISTIO_PROXY_OPENSSL_GIT_REPO:=https://github.com/maistra/istio-proxy-openssl}
```
https://unix.stackexchange.com/questions/25425/what-does-param-value-mean

Reply:
ya, that's cleaner